frontend/src/components/enhanced-markdown.tsx

```
@@ -71,13 +73,23 @@ const CodeComponent = ({ inline, className, children }: any) => {
         components={{
-          code: (props) => <CodeComponent {...props} />,
+          code: ({ className, children, ...props }) => {
+            const codeString = String(children || '');
+            const match = /language-(\w+)/.exec(className || '');
+            const language = match ? match[1] : 'text';
+
+            if ( language === 'mermaid') {
+              return <MermaidChart chart={codeString} className="my-4 flex justify-center" />;
+            }
+
+            return <CodeComponent className={className} children={children} {...props} />;
```

Biome (2.1.2)

[error] Avoid passing children using a prop

The canonical way to pass children in React is to use JSX elements

(lint/correctness/noChildrenProp)